demos/multi_material/thermochemical_buoyancy/van_keken_1997.py

```diff
@@ -0,0 +1,277 @@
+# Rayleigh-Taylor instability
+# ===========================
+#
+# Rationale
+# ---------
+#
+# Our previous tutorial introduced multi-material simulations in G-ADOPT by
+# investigating compositional effects on buoyancy. We extend that tutorial to include
+# thermal effects, thereby simulating thermochemical convection, which is, for example,
+# essential to modelling Earth's mantle evolution.
+#
+# This example
+# ------------
+#
+# Here, we consider the entrainment of a thin, compositionally dense layer by thermal
+# convection presented in van Keken et al. (1997). Inside a 2-D domain heated from
+# below, a denser material sits at the bottom boundary beneath a lighter material.
+# Whilst the compositional stratification is stable, heat transfer from the boundary
+# generates positive buoyancy in the denser material, allowing thin layers of material
```

Comment:
Instead of thing layers, perhaps use thin "tendrils" - that's the lingo I seem to remember.

Reply:
I would add a comment at the end of this paragraph that the goal of this tutorial is to illustrate how to set up such a problem. To make the tutorial tractable, it must be run at a coarser resolution than is required, and hence results differ to those in the benchmark. Nonetheless, you have verified, by simulating this case at higher resolution, that things do converge towards those benchmark values.